lib/internal/webstreams/writablestream.js

```diff
@@ -1263,18 +1278,35 @@ function setupWritableStreamDefaultController(
     PromiseResolve(startResult),
     () => {
       assert(stream[kState].state === 'writable' ||
-             stream[kState].state === 'erroring');
+        stream[kState].state === 'erroring');
       controller[kState].started = true;
       writableStreamDefaultControllerAdvanceQueueIfNeeded(controller);
     },
     (error) => {
       assert(stream[kState].state === 'writable' ||
-             stream[kState].state === 'erroring');
+        stream[kState].state === 'erroring');
       controller[kState].started = true;
       writableStreamDealWithRejection(stream, error);
     });
 }
 
+function writableStreamAddFinishedCallback(stream, callback) {
+  assert(isWritableStream(stream));
+  stream[kState].finishedCallbacks.push(callback);
+}
+
+function writableStreamRemoveAllFinishedCallbacks(stream) {
+  assert(isWritableStream(stream));
+  stream[kState].finishedCallbacks = [];
+}
+
+function writableStreamCallAllFinishedCallbacks(stream) {
+  assert(isWritableStream(stream));
+  stream[kState].finishedCallbacks?.forEach((cb) => {
+    cb(stream[kState].storedError);
+  });
+}
```

Comment:
I think this is the wrong approach. the `WritableStream` and `ReadableStream` APIs already have mechanisms for monitoring when each are closed via the `closed` promises on the `Writer` and `Reader`.